quartz_solar_forecast/data.py

```diff
@@ -37,16 +38,39 @@ def get_gfs_nwp(site: PVSite, ts:datetime) -> xr.Dataset:
     start = ts.date()
     end = start + pd.Timedelta(days=7)
 
-    # Getting GFS, from OPEN METEO
+    # Getting NWP, from OPEN METEO
+    urlStart = None
+    if source == "icon":
+        urlStart = "dwd-icon"
+    if source == "gfs":
+        urlStart = "gfs"
+
     url = (
-        f"https://api.open-meteo.com/v1/gfs?"
+        f"https://api.open-meteo.com/v1/{urlStart}?"
         f"latitude={site.latitude}&longitude={site.longitude}"
         f"&hourly={','.join(variables)}"
         f"&start_date={start}&end_date={end}"
     )
     r = requests.get(url)
     d = json.loads(r.text)
 
+    # If the source is icon, add gfs visibility data
+    if source == "icon":
+        url = (
+            f"https://api.open-meteo.com/v1/gfs?"
+            f"latitude={site.latitude}&longitude={site.longitude}"
+            f"&hourly=visibility"
+            f"&start_date={start}&end_date={end}"
+        )
+        r_gfs = requests.get(url)
+        d_gfs = json.loads(r_gfs.text)
+
+        # get visibility data
```

Comment:
could add something here to the comments, saying
`get visibility data from GFS as its not available with ICON`